js/ui/calendar/ui.calendar.selection.strategy.js

```diff
@@ -0,0 +1,62 @@
+class CalendarSelectionStrategy {
+    constructor(component) {
+        this.calendar = component;
```

Comment:
I am believe that strategy object shouldn't know anything about component and must not manipulate state of the component.

Reply:
I agree with u, but the actual approach is already used in all out strategies and it's not a fast task to separate a responsibility between calendar and its strategies now.

I suggest to note this in spec / card, and postpone this improvement.

Reply:
We discussed it in voice. We agree with u, but decided to postpone this refactoring. It was implemented this way because all strategies in DevExtreme uses parent component instance.